src/components/refactor/utils/yup-config.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import * as yup from 'yup';
+
+yup.setLocale({
+    mixed: {
+        required: 'YupRequired',
+        notType: ({ type }) => {
+            if (type === 'number') {
+                return 'YupNotTypeNumber';
+            } else {
+                return 'YupNotTypeDefault';
+            }
+        },
+    },
+});
+
+// yup does not support empty number fields, it returns typeError
+// this methods returns an error if the string is not empty and is not a number
+yup.addMethod(yup.string, 'nullableNumber', function (message) {
```

Comment:
because this is non standard and doesn't autocomplete, maybe it's better to handle in the transform of react-hook-form:

then

number().nullable().required() has the following behavior:
```
empty =>  "XXX is a required field",
foo => "firstName must be a `number` type, but the final value was: `NaN` (cast from the value `\"foo\"`).
123 => OK
```

number().nullable() has the following behavior:
```
empty =>  "XXX is a required field",
foo => "firstName must be a `number` type, but the final value was: `NaN` (cast from the value `\"foo\"`).
123 => OK
```

Reply:
fixed, added transforminput / transformOuput in order to do this